SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -390,37 +405,188 @@ void USpatialNetDriverGameplayDebuggerContext::OnServerRequest(AGameplayDebugger
+		EntityViewPtr->Components.FindByPredicate([](const SpatialGDK::ComponentData& Data) {
+			return Data.GetComponentId() == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID;
+		});
+
+	if (IntentComponentData == nullptr)
+	{
+		return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+	}
+
+	SpatialGDK::AuthorityIntent Intent = SpatialGDK::AuthorityIntent(IntentComponentData->GetUnderlying());
+	const VirtualWorkerId VirtualWorkerId = Intent.VirtualWorkerId;
+	return VirtualWorkerId;
+};
+
+void USpatialNetDriverGameplayDebuggerContext::RegisterPlayerControllerAuthorityLostCallback(
+	AGameplayDebuggerCategoryReplicator& InReplicator, FEntityData& InEntityData)
+{
+	if (!InEntityData.PlayerControllerAuthorityChangeHandle.IsValid())
+	{
+		if (APlayerController* PlayerControllerActor = Cast<APlayerController>(InReplicator.GetReplicationOwner()))
+		{
+			InEntityData.PlayerControllerAuthorityChangeHandle = PlayerControllerActor->OnAuthorityLostDelegate().AddUObject(
+				this, &USpatialNetDriverGameplayDebuggerContext::OnPlayerControllerAuthorityLost);
 		}
-		else
+	}
+	else
+	{
+		UE_LOG(LogSpatialNetDriverGameplayDebuggerContext, Error, TEXT("Trying to bind change notification more than once"));
+	}
+}
+
+void USpatialNetDriverGameplayDebuggerContext::UnregisterPlayerControllerAuthorityLostCallback(
+	AGameplayDebuggerCategoryReplicator& InReplicator, FEntityData& InEntityData)
+{
+	check(NetDriver && NetDriver->PackageMap);
+
+	if (InEntityData.PlayerControllerAuthorityChangeHandle.IsValid())
+	{
+		if (APlayerController* PlayerController = InReplicator.GetReplicationOwner())
+		{
+			PlayerController->OnAuthorityLostDelegate().Remove(InEntityData.PlayerControllerAuthorityChangeHandle);
+			InEntityData.PlayerControllerAuthorityChangeHandle.Reset();
+		}
+	}
+}
+
+void USpatialNetDriverGameplayDebuggerContext::OnPlayerControllerAuthorityLost(const APlayerController& InPlayerController)
+{
+	check(NetDriver && NetDriver->PackageMap);
+
+	Worker_EntityId_Key* EntityId = nullptr;
+	FEntityData* EntityData = nullptr;
+	AGameplayDebuggerCategoryReplicator* Replicator = nullptr;
+
+	for (auto& TrackedEntity : TrackedEntities)
+	{
+		Replicator = TrackedEntity.Value.ReplicatorWeakObjectPtr.Get();
+		if (Replicator != nullptr)
 		{
-			UE_LOG(LogSpatialNetDriverGameplayDebuggerContext, Error, TEXT("Callback from an actor to unknown virtual worker"));
+			if (Replicator->GetReplicationOwner() == &InPlayerController)
+			{
+				EntityId = &TrackedEntity.Key;
+				EntityData = &TrackedEntity.Value;
+			}
 		}
 	}
+
+	if (EntityId == nullptr || EntityData == nullptr || Replicator == nullptr)
+	{
+		return;
+	}
+
+	if (EntityData->Component.TrackPlayer == false)
+	{
+		// If we are not tracking the player, we do not need to track server authority changes
+		return;
+	}
+
+	check(NetDriver->LoadBalanceStrategy);
+	const VirtualWorkerId AuthorativeWorkerId = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(InPlayerController);
```

Comment:
Did you used to look at the AuthorityIntent component value here, or did I imagine that?